circuit/src/utils.rs

```diff
@@ -0,0 +1,151 @@
+use plonky2::field::types::{Field, Field64, PrimeField64};
+use crate::circuit::F;
+
+pub fn u128_to_felts(num: u128) -> Vec<F> {
+    let mut amount_felts: Vec<F> = Vec::with_capacity(2);
+    let amount_high = F::from_noncanonical_u64((num >> 64) as u64 % F::ORDER);
+    let amount_low =  F::from_noncanonical_u64(num as u64 % F::ORDER);
```

Comment:
Do we need the `... % F::ORDER` after all?

Reply:
no I don't think so, removed